pkg/apis/provisioning/v1alpha5/constraints.go

```diff
@@ -47,11 +47,20 @@ type Constraints struct {
 	// Provider contains fields specific to your cloudprovider.
 	// +kubebuilder:pruning:PreserveUnknownFields
 	Provider *Provider `json:"provider,omitempty"`
+	// ProviderRef is a reference to a dedicated CRD for the chosen provider, that holds
+	// additional configuration options
+	// +optional
+	ProviderRef *ProviderRef `json:"providerRef,omitempty"`
 }
 
 // +kubebuilder:object:generate=false
 type Provider = runtime.RawExtension
 
+type ProviderRef struct {
```

Comment:
Thoughts on reusing an existing struct? https://pkg.go.dev/k8s.io/api/autoscaling/v1#CrossVersionObjectReference. This may be a bad choice since it's from `autoscaling` api group, but let me look for others.

Reply:
Hmm -- maybe this is bad advice. There are multiple places where this is defined similarly. Perhaps we can just copy the syntax? 

https://github.com/kubernetes/api/search?l=Go&q=APIVersion

```
	// Kind of the referent; More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds"
	Kind string `json:"kind" protobuf:"bytes,1,opt,name=kind"`
	// Name of the referent; More info: http://kubernetes.io/docs/user-guide/identifiers#names
	Name string `json:"name" protobuf:"bytes,2,opt,name=name"`
	// API version of the referent
	// +optional
	APIVersion string `json:"apiVersion,omitempty" protobuf:"bytes,3,opt,name=apiVersion"`
```

Reply:
Changed it to this ^. I like being able to control what `Kind` and `APIVersion` are described as in the CRD so inlined rather than using typeMeta.